homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -157,12 +199,17 @@ async def async_added_to_hass(self) -> None:
             )
         )
         _LOGGER.debug(
-            "Setup of price sensor %s (%s) with tariff '%s'",
-            self.name,
+            "Setup of ESIOS sensor %s (%s, unique_id: %s)",
+            self.entity_description.key,
             self.entity_id,
-            self.coordinator.api.tariff,
+            self._attr_unique_id,
         )
 
+    @callback
+    def async_registry_entry_updated(self) -> None:
+        """Enable API downloads for this sensor."""
+        self.coordinator.api.update_active_sensors(self.entity_description.key, True)
```

Comment:
What is this meant to do?

Reply:
The external API handler is shared, and it tries to download prices periodically for each day and sensor. For disabled sensors, there is no need to download, so there is where they get flagged for download. Should I make this in another way?

Reply:
This callback is called whenever this sensor's entity registry entry is updated, it doesn't seem correct to assume an update of the registry entry means the sensor needs updates.
I'd move this call to `async_added_to_hass` instead.
Also, this should be called from `async_added_to_hass`:
```py
        self.async_on_remove(
            lambda: coordinator.api.update_active_sensors(
                self.entity_description.key, False
            )
        )
```

Reply:
> is called whenever this sensor's entity registry entry is updated

Oh, I missed that! I noticed that method was called when enabling disabled sensors, and never if the sensor was disabled, so it looked like a good place to place that.

The `async_on_remove` callback is already defined, but in the sensor's init. 

**I will move both to `async_added_to_hass` asap** 👍

Reply:
Little problem @emontnemery : 

if I enable the sensors inside the `async_added_to_hass`, all sensors are enabled, not following the `entity_registry_enabled_default` flag...

```python
    async def async_added_to_hass(self) -> None:
        """Handle entity which will be added."""
        await super().async_added_to_hass()

        # Enable API downloads for this sensor
        self.coordinator.api.update_active_sensors(self.entity_description.key, True)
        self.async_on_remove(
            lambda: self.coordinator.api.update_active_sensors(
                self.entity_description.key, False
            )
        )

        # Update 'state' value in hour changes
        self.async_on_remove(
            async_track_time_change(
                self.hass, self.update_current_price, second=[0], minute=[0]
            )
        )
```

I think this is why I put it in `async_registry_entry_updated`. Should I query `self.registry_entry` to enable downloads for each sensor only for when it's not disabled?

Reply:
`async_added_to_hass` is not called if the `entity_registry_enabled_default` is set to `False` unless the user enables the entity.

I think this is a problem related to testing, the `entity_registry_enabled_default` is only considered when the config entry is first added.

Please double check this

Reply:
> if I enable the sensors inside the async_added_to_hass, all sensors are enabled

The problem was in my logic to retrieve the 'disabled sensors' when running `async_setup_entry` 🙈. Now it's fixed, to apply defaults in the first activation of the API token, or check the .disabled attribute in other case. 

Thanks for pointing it out 👍